TempleEight/docs/guide/orchestration.md

```diff
@@ -3,3 +3,172 @@ id: orchestration
+
+```templefile {4}
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+We can see here that the Templefile features a `#provider` annotation in the project block.
+This is how we specify that our project uses orchestration, the name in the argument specifies which framework we are generating for.
+
+The currently supported frameworks are:
+
+* Docker Compose
+
+* Kubernetes
+
+All of the frameworks are generated to share a common toolchain, so that usage is the same no matter which you choose.
+
+## Kong API Gateway
+
+An API Gateway is an infrastructure component designed to be the entry point to your application. 
+It receives all requests from the user, performs actions like verifying their authentication, and then forwards the request to the correct microservice.
+
+Temple makes use of the existing [Kong API Gateway](https://docs.konghq.com/2.0.x/getting-started/quickstart/) for this purpose, and automatically generates all the configuration it requires.
+
+The only considerations you as the user needs to make is that when you try to make requests to any of your deployed services, the requests need to be addressed to Kong's URL.
+Temple's tooling automatically sets this URL into an environment variable called `$KONG_ENTRY`. 
+
+As previously mentioned, Kong also handles some end-user authentication, when it's used in your project.
+See the [Authentication](authentication) Guide for full details of this.
+
+## Docker Compose
+
+[Docker Compose](https://docs.docker.com/compose/) is a tool built into the docker ecosystem for orchestrating containers.
+We recommend using it for local development, as running your services this way has comparatively little overhead.
+However it doesn't allow for some more advanced features like automatically replicating and distributing services across different machines out of the box.
+
+When the `dockerCompose` provider is selected, Temple generates three important files, namely:
+
+```shell
+.
+├── deploy.sh
+├── docker-compose.yml
+└── kong
+    └── configure-kong.sh
+```
+
+First, we'll talk about `docker-compose.yml`.
+This file acts to instruct Docker-Compose on how to manage your services. 
+It specifies, for each service, which Docker image should be used in the container, any volume mounts, and environment variables needed.
+It also defines networking, allowing certain services to speak to others (for example, only one service should be able to communicate with each database).
+
+`kong/configure-kong.sh` is a script that is ran once the docker-compose infrastructure is running.
+It sends a series of `cURL` requests to the `Kong` API gateway which configures it to route requests it receives to the correct service, and to deal with any auth required 
+(see the [Auth](authentication) Guide, and the [Kong Documentation](https://docs.konghq.com/2.0.x/getting-started/quickstart/)).
```

Comment:
Authentication over just auth, as it could mean Authorisation, which isn't the same
```suggestion
(see the [Authentication](authentication) Guide, and the [Kong Documentation](https://docs.konghq.com/2.0.x/getting-started/quickstart/)).
```